### ISSUE TITLE
Misaligned Windows

### ISSUE THREAD

**Issue by zugno** [qtile/qtile#3501]
### The issue:

Good day,
I have noticed a window misalignment issue with certain configurations of certain layouts. This issue seems to be affected by the height of the bar.  I am going to provide some screenshots using different variations of the default config.
My system specs are:

- Qtile version: 0.21.0
- Distro: Arch Linux
- Display Server: Xorg
- Display resolution: 2560 x 1440

In the screenshots I used "margin = 3" in the MonadTall and Columns layouts in order to make the issue more evident, but the issue is present also with no margin.

- MonadTall, even bar height:
![MonadTall-BarEven](https://user-images.githubusercontent.com/66468432/162983551-ec9e0225-32e5-44ad-acbc-d91f6a275508.png)
![MonadTall-BarEven-Macro](https://user-images.githubusercontent.com/66468432/162984675-81910c95-3e15-426c-a59a-31cb04c91d6e.png) Bottom-most right window is 1px shorter than it should.

- MonadTall, odd bar height:
![MonadTall-BarOdd](https://user-images.githubusercontent.com/66468432/162984023-2c9e7550-bdb7-41c1-aea3-961d7e83d008.png)
![MonadTall-BarOdd-Macro](https://user-images.githubusercontent.com/66468432/162984702-d8d88f59-bf93-411d-850d-8346bd0cbcf5.png) Bottom-most right window is 1px shorter than it should.

- Columns, odd bar height:
![Columns-BarOdd](https://user-images.githubusercontent.com/66468432/162985781-b7518de8-aebe-4727-bb9a-42d539edcfc4.png)
![Columns-BarOdd-Macro](https://user-images.githubusercontent.com/66468432/162985793-e3eecb80-8955-4c58-a8b8-f73a12421092.png) Bottom-most right window is 1px taller than it should.

- MonadWide:
![MonadWide-BarOE](https://user-images.githubusercontent.com/66468432/162985251-2f443e94-b9ee-4d75-a91c-7e5542f7f4e4.png)
Here, even with no margin specified, there is a 1px gap on both sides of the second and last bottom windows.

Thank you for your work.

### Required:

- [X] I have searched past issues to see if this bug has already been reported.

**Comment by m-col** [comment]
Yeah I've seen this, it's not ideal. I think it affects a few other layouts as well, and it's just like you describe @elParaguayo.

I think the layouts should be able to account for this. Having the rounding errors increase the gap sizes is much more noticable than putting them in the window widths/heights.

**Comment by mbenford** [comment]
I remember spending some time making sure all heights, widths and margins were "correct" in MonadThreeCol. Maybe we can move that logic into the base layout so all layouts can benefit from it?

![image](https://user-images.githubusercontent.com/446867/163595745-056046fa-8ee8-4462-bd6b-0800b7772318.png)

![image](https://user-images.githubusercontent.com/446867/163595834-c6619d51-dce5-4104-a2a3-0d869f9ef3da.png)

I can try to work on that along with #3337.

**Comment by zugno** [comment]
I did a quick test and, from what I could see, I can confirm that there is no misalignment in the MonadThreeCol layout.

**Comment by zugno** [comment]
Bumping so the issue will not be automatically closed.

**Comment by clotodex** [comment]
Another thought, do you have a HiDpi screen? Since qtile right now can not define layouts and bar etc on scaled outputs https://github.com/qtile/qtile/issues/3646

**Comment by zugno** [comment]
> Another thought, do you have a HiDpi screen? Since qtile right now can not define layouts and bar etc on scaled outputs #3646

I don't think a standard 2560x1440 (QHD) can be defined as HiDPI
